File: test-suite/src/test/java/io/micronaut/docs/factories/primitive/EngineSpec.java
package io.micronaut.docs.factories.primitive;

import io.micronaut.context.ApplicationContext;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class EngineSpec {
    @Test
    void testEngine() {
        try (ApplicationContext context = ApplicationContext.run()) {
            final V8Engine engine = context.getBean(V8Engine.class);
            assertEquals(
                    8,
                    engine.getCylinders()
            );
        }
    }
}

File: test-suite-scala/src/test/scala/io/micronaut/docs/factories/primitive/EngineSpec.scala
package io.micronaut.docs.factories.primitive

import io.micronaut.context.ApplicationContext
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class EngineSpec:

  @Test
  def testEngine(): Unit =
    val context = ApplicationContext.run()
    try
      val engine = context.getBean(classOf[V8Engine])
      assertEquals(8, engine.getCylinders())
    finally context.close()
